db/blob/blob_source.cc

```diff
@@ -152,6 +153,142 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
   return s;
 }
 
+void BlobSource::MultiGetBlob(
+    const ReadOptions& read_options,
+    const autovector<std::reference_wrapper<const Slice>>& user_keys,
+    uint64_t file_number, uint64_t file_size,
```

Comment:
I assume we will support `MultiGetBlob()` from multiple files?

Reply:
I think MultiGetBlob() is to get multiple blobs from same file. For multiple files, we have an upper level function for that in VersionSet: https://github.com/facebook/rocksdb/blob/a16e2ff82ae77886a6f8cd5cb5d8bdec66c486bd/db/version_set.cc#L1931

Version::MultiGetBlob will call  BlobSource::MultiGetBlob multiple times.

Reply:
That's actually a very good question, shall we put Version::MultiGetBlob's logic into a new Version::MultiGetBlob?

Reply:
+1 for this. I think this is another piece that could be pushed down to `BlobSource`. (Well, most of it I guess: it would make sense to keep e.g. the blob file number validity check in `Version`.)

Reply:
All right. Let's make a PR for cleanup version and add new api `MultiGetBlobFromOneFile`.  I don't want to do another huge PR.

Reply:
Added: https://github.com/facebook/rocksdb/issues/10156#issuecomment-1163469932